doc/feature/httpServerREST.md

```diff
@@ -0,0 +1,161 @@
+HTTP Server for generic REST interfaces {#httpserverrest}
+============
+
+This document describes a proposed extension to the current http::Server to make
+it usable as a general-purpose REST interface.
+
+## Requirements
+
+* Support all standard http verbs: GET, POST, PUT, PATCH, DELETE
+* Support custom response payload for all action verbs (not just for GET)
+* Allow specifying the MIME type of the response to serve any type of content
+* Allow specifying extra HTTP header fields in response to certain requests
+* Allow asynchronous processing of requests (not blocking the main thread)
+* Single endpoint for collections (e.g. windows/, files/, neurons/, etc. )
+* Allow filtering operations on collections (windows/selected, files?limit=10)
+
+## Dependency Changes
+
+None, cppnetlib 0.13 already offers the required functionality.
+
+## API
+
+To fullfill the abovementioned requirements with minimal API changes, we propose
+a generic interface as follows:
+
+    namespace zeroeq
+    {
+    namespace http
+    {
+    enum class Verb { GET, POST, PUT, PATCH, DELETE };
```

Comment:
True, but I must say that after using both for a while I still find that upper-case for constants/enum makes the code easier to read. That's just a detail though.